zk/lock.go

```diff
@@ -43,6 +46,12 @@ func parseSeq(path string) (int, error) {
 // is acquired or an error occurs. If this instance already has the lock
 // then ErrDeadlock is returned.
 func (l *Lock) Lock() error {
+	if(mapEphermeralSequenceLockPath[l.path]!="") {
```

Comment:
i beleive it will be more go way to right this like this:
```
if   zPath, ok := mapEphermeralSequenceLockPath[l.path]; ok && zPath != "" {
	if lockExists(l.c, l.path, zPath)) {
			return nil
		}
}
```

Reply:
Agreed. Please check